src/Client/OfficeInterop/OfficeInterop.fs

```diff
@@ -798,31 +853,206 @@ let replaceOutputColumn (annotationTableName:string) (existingOutputColumn: Buil
+    let possibleInputColumn = arcTable.TryGetInputColumn()
+    let inputColumnName =
+        if possibleInputColumn.IsSome then possibleInputColumn.Value.Header.ToString()
+        else ""
 
-            let! existingBuildingBlocks = BuildingBlock.getFromContext(context,annotationTable)
+    let columns = annotationTable.columns
+    let inputColumn =
+        columns.items
+        |> Array.ofSeq
+        |> Array.tryFind(fun col -> col.name = inputColumnName)
 
-            // comment out, to reenable inserting multiple duplicate building block
-            //checkIfBuildingBlockExisting newBB existingBuildingBlocks
+    if inputColumn.IsSome then
 
-            checkHasExistingInput newBB existingBuildingBlocks
-            checkIfBuildingBlockExisting newBB existingBuildingBlocks
-            // if newBB is output column and output column already exists in table this returns (Some outputcolumn-building-block), else None.
-            let outputColOpt = checkHasExistingOutput newBB existingBuildingBlocks
+        let oldInputColumn = annotationTable.columns.items.[(int) inputColumn.Value.index].name
 
-            let! res = 
-                match outputColOpt with
-                | Some existingOutputColumn -> replaceOutputColumn annotationTableName existingOutputColumn newBB 
-                | None -> addAnnotationBlock newBB
+        //Only update the input column when it has a new value
+        if oldInputColumn <> newBB.Header.ToString() then
+            annotationTable.columns.items.[(int) inputColumn.Value.index].name <- newBB.Header.ToString()
 
-            return res
+        let warningMsg =
+            if oldInputColumn = newBB.Header.ToString() then
+                $"Found existing input column \"{oldInputColumn}\". Did not change the column because the new input column is the same \"{annotationTable.columns.items.[(int) inputColumn.Value.index].name}\"."
+            else                
+                $"Found existing input column \"{oldInputColumn}\". Changed input column to \"{annotationTable.columns.items.[(int) inputColumn.Value.index].name}\"."
+
+        let msg = InteropLogging.Msg.create InteropLogging.Warning warningMsg
+
+        let loggingList = [
+            msg
+        ]
+
+        loggingList
+    else
+        failwith "Something went wrong! The update input column is not filled with data! Please report this as a bug to the developers."
+
+/// <summary>
+/// Add a new inputcolumn to an annotation table
+/// </summary>
+/// <param name="annotationTable"></param>
+/// <param name="arcTable"></param>
+/// <param name="newBB"></param>
+let addInputColumn (annotationTable:Table) (arcTable:ArcTable) (newBB:CompositeColumn) =
+
+    if arcTable.TryGetInputColumn().IsSome then
+        failwith "Something went wrong! The add input column is filled with data! Please report this as a bug to the developers."
+
+    else
+
+        let rowCount = arcTable.RowCount + 1
+
+        let createCol name =
```

Comment:
Maybe make this generic and reuseable for other functions

Reply:
fixed